samcli/local/docker/lambda_build_container.py

```diff
@@ -0,0 +1,164 @@
+        manifest_file_name = abs_manifest_path.name
+        manifest_dir = str(abs_manifest_path.parent)
+
+        source_dir = str(pathlib.Path(source_dir).resolve())
+
+        container_dirs = LambdaBuildContainer._get_container_dirs(source_dir, manifest_dir)
+
+        request_json = self._make_request(protocol_version,
+                                          language,
+                                          dependency_manager,
+                                          application_framework,
+                                          container_dirs,
+                                          manifest_file_name,
+                                          runtime,
+                                          optimizations,
+                                          options)
+
+        image = LambdaBuildContainer._get_image(runtime)
+        entry = LambdaBuildContainer._get_entrypoint(request_json)
+        cmd = []
+
+        additional_volumes = {
+            # Manifest is mounted separately in order to support the case where manifest
+            # is outside of source directory
+            manifest_dir: {
+                "bind": container_dirs["manifest_dir"],
+                "mode": "ro"
+            }
+        }
+
+        env_vars = None
+        if log_level:
+            env_vars = {
+                "LAMBDA_BUILDERS_LOG_LEVEL": log_level
+            }
+
+        super(LambdaBuildContainer, self).__init__(
+            image,
+            cmd,
+            container_dirs["source_dir"],
+            source_dir,
+            additional_volumes=additional_volumes,
+            entrypoint=entry,
+            env_vars=env_vars)
+
+    @property
+    def executable_name(self):
+        return LambdaBuildContainer._BUILDERS_EXECUTABLE
+
+    @staticmethod
+    def _make_request(protocol_version,
+                      language,
+                      dependency_manager,
+                      application_framework,
+                      container_dirs,
+                      manifest_file_name,
+                      runtime,
+                      optimizations,
+                      options):
+
+        return json.dumps({
+            "jsonschema": "2.0",
+            "id": 1,
+            "method": "LambdaBuilder.build",
+            "params": {
+                "__protocol_version": protocol_version,
+                "capability": {
+                    "language": language,
+                    "dependency_manager": dependency_manager,
+                    "application_framework": application_framework
+                },
+                "source_dir": container_dirs["source_dir"],
+                "artifacts_dir": container_dirs["artifacts_dir"],
+                "scratch_dir": container_dirs["scratch_dir"],
+
+                # Path is always inside a Linux container. So '/' is valid
+                "manifest_path": "{}/{}".format(container_dirs["manifest_dir"], manifest_file_name),
```

Comment:
we can use posixpath.sep or something like that too, if we dont want to use '/' explicitly.

Reply:
ah right! I wanted to be explicit here so someone doesn't accidentaly "fix" it later